datafusion/sql/src/unparser/plan.rs

```diff
@@ -157,10 +157,95 @@ impl Unparser<'_> {
             )]);
         }
 
+        // Construct a list of all the identifiers present in query sources
+        let mut all_idents = Vec::new();
```

Comment:
Wondering if this can be wrapped to a helper function like `collect_valid_idents` or similar?